src/rc.c

```diff
@@ -153,4 +162,66 @@ bool receive_rc(uint32_t now)
   return true;
 }
 
+void calibrate_rc()
+{
+  if(_armed_state == ARMED)
+  {
+    mavlink_log_error("Cannot calibrate RC when FCU is armed");
+  }
+  else
+  {
+    // Calibrate Extents of RC Transmitter
+    mavlink_log_warning("Calibrating RC, move sticks to full extents");
+    mavlink_log_warning("in the next 10s");
+    uint32_t now = micros();
+    static int32_t max[4] = {0, 0, 0, 0};
+    static int32_t min[4] = {10000, 10000, 10000, 10000};
+    while(micros() - now < 1e7)
+    {
+      for(int16_t i = 0; i < 4; i++)
+      {
+        int32_t read_value = (int32_t)pwmRead(i);
+        if(read_value > max[i])
+        {
+          max[i] = read_value;
+        }
+        if(read_value < min[i])
+        {
+          min[i] = read_value;
+        }
+      }
+      delay(10);
+    }
+    _params.values[PARAM_RC_X_RANGE] = max[_params.values[PARAM_RC_X_CHANNEL]] - min[_params.values[PARAM_RC_X_CHANNEL]];
```

Comment:
I think that we can require that the range is equally spaced about the center.  (It would be pretty weird to have something else).

I have a pull request pending (rc_switch branch) that builds a switch_t with a direction.  As for the actual channels, the transmitter is a more convenient place to have the direction switch.  (I used to have a direction switch for each, but then I messed it up so many times that I just decided to leave it on the transmitter, rather than dealing with it in code.)